### ISSUE TITLE
nesting profiler bug

### ISSUE THREAD

**Issue by NoriteSC** [FlaxEngine/FlaxEngine#1380]
**Issue description:**

nesting profiler calls breaks it the window
**Steps to reproduce:**

**Minimal reproduction project:**

``` cs
void Update()
{
Profiler.BeginEvent("A");
//some code
Profiler.BeginEvent("B");
//some code
Profiler.EndEvent();
//some code
Profiler.EndEvent();
}
```

**Comment by GoaLitiuM** [comment]
Seems to be an oversight with the Profiler API. The C#-methods for named events are not handling the returned index parameter correctly. `BeginEvent("A")` should return the index which is supposed to be ended by passing the index to the `EndEvent`-method.

**Comment by NoriteSC** [comment]
@GoaLitiuM yes looks like it
but this can be 100% automatic unless there is a need to run profiler between more than 1 frame

**Comment by mafiesto4** [comment]
Fixed in 0eda67bd738dbb061a9214700bb1d5a0626ea362.
